libs/utils/utils.js

```diff
@@ -144,6 +144,27 @@ export const [setConfig, getConfig] = (() => {
         // eslint-disable-next-line no-console
         console.log('Invalid or missing locale:', e);
       }
+      if (Array.isArray(config.contentRoot)) {
+        const pathSplit = pathname.split('/'); 
+        const contentRoots = config.contentRoot;   
+        // page is in contentRoot array, is in en-US.
+        if (contentRoots.includes(pathSplit[1])) {
+          config.contentRoot = `/${pathSplit[1]}`;
+        } else if (contentRoots.includes(pathSplit[2])) {
+          if (config.locale.prefix.length > 0) {
+            config.contentRoot = `/${pathSplit[2]}`;
+          } else {
+            // second split is present, but en-US prefix ie: /au/sub-cat.
+            config.contentRoot = `/${pathSplit[1]}/${pathSplit[2]}`;
+          }
+        } else if (!config.locale.prefix.length > 0 && !contentRoots.includes(pathSplit[1]) && pathSplit[2]) {
+          // page is not in contentRoot array, has no prefix (us-en), and isn't root index. ie: root page of /au/ site.
+          config.contentRoot = `/${pathSplit[1]}`;
+        } else {
+          // Fallback - ie: "/".
+          config.contentRoot = '';
+        }
+      }
       config.locale.contentRoot = `${origin}${config.locale.prefix}${config.contentRoot ?? ''}`;
```

Comment:
With your changes, will this become `/de/de/contentroooot` ?

Reply:
I don't think so. This one comes out to /kr/acrobat for example. 
https://dynamic-gnav--fedpub--adobecom.hlx.page/kr/acrobat/hub/how-to/10-benefits-of-digitizing-documents